Tests/Bridge/Exporter/AdminExporterTest.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Bridge\Exporter;
+
+use Exporter\Exporter;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Bridge\Exporter\AdminExporter;
+
+class AdminExporterTest extends \PHPUnit_Framework_TestCase
+{
+    public function provideExportFormats()
+    {
+        return array(
+            'no override' => array(array('xls'), AbstractAdmin::DEFAULT_EXPORT_FORMATS, array('xls')),
+            'override in admin' => array(array('csv'), array('csv'), array('xls')),
+        );
+    }
+
+    /**
+     * @dataProvider provideExportFormats
+     */
+    public function testAdminHasPriorityOverGlobalSettings(array $expectedExportFormats, array $adminExportFormats, array $globalExportFormats)
+    {
+        $writers = array();
+        foreach ($globalExportFormats as $exportFormat) {
+            $writer = $this->getMock('Exporter\Writer\TypedWriterInterface');
+            $writer->expects($this->once())
+                ->method('getFormat')
+                ->will($this->returnValue($exportFormat));
+            $writers[] = $writer;
+        }
+        $exporter = new Exporter($writers);
```

Comment:
I can't mock this class b/c it is final. I think I should have created an interface for that…